### PR TITLE
feat(Core/Common): delete lib game-interface inherited

### PR DESCRIPTION
**Pull request by Winfidonarleyan**
## Changes Proposed:
- https://github.com/WarheadCore/Warhead/commit/130878c1a5d3f1ae42564a14fb6f0a19840f8749
- https://github.com/WarheadCore/Warhead/commit/67f9d916cbd920ee72b3e9036e13d0725b38566e
- https://github.com/WarheadCore/Warhead/commit/aa042e94c2f76e5735a8f745c3fcb7dd54cf6c0b
- https://github.com/WarheadCore/Warhead/commit/ca7f31c6cdceb7e2921d5d83e1ae22ccb247988d
- This pr not changes game, only delete inherited from library `game-interface`

## Issues Addressed:
- Part #4302
- Part #4669 

## Tests Performed:
- In game with 30+ min uptime

## How to Test the Changes:
- I think just test maps / instances and etc
- Try test last changes, example #4922

## Target Branch(es):
- [x] Master

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/common/Collision/Maps/MapTree.cpp [hunk 1]
```diff
@@ -462,4 +462,10 @@ namespace VMAP
         }
         iLoadedTiles.erase(tile);
     }
+
+    void StaticMapTree::getModelInstances(ModelInstance*& models, uint32& count)
```
Comment (Winfidonarleyan) [inline src/common/Collision/Maps/MapTree.cpp:466]: .

File: src/common/Collision/Management/VMapManager2.cpp [hunk 2]
```diff
@@ -166,7 +166,7 @@ namespace VMAP
     bool VMapManager2::isInLineOfSight(unsigned int mapId, float x1, float y1, float z1, float x2, float y2, float z2)
     {
 #if defined(ENABLE_VMAP_CHECKS)
-        if (!isLineOfSightCalcEnabled() || !IsVMAPDisabledForPtr(mapId, VMAP_DISABLE_LOS))
+        if (!isLineOfSightCalcEnabled() || IsVMAPDisabledForPtr(mapId, VMAP_DISABLE_LOS))
```
Comment (FrancescoBorzi) [inline src/common/Collision/Management/VMapManager2.cpp:169]: how is this change related to the title of this PR?

Comment (Winfidonarleyan) [reply src/common/Collision/Management/VMapManager2.cpp:169]: `DisableMgr::IsDisabledFor` to `IsVMAPDisabledForPtr` ?

Comment (Winfidonarleyan) [reply src/common/Collision/Management/VMapManager2.cpp:169]: General change - delete `game-interface` from `common` lib
image

Other changes - support build without game lib

Comment (FrancescoBorzi) [reply src/common/Collision/Management/VMapManager2.cpp:169]: I meant why removing `!` from `!IsVMAPDisabledForPtr`

Comment (Winfidonarleyan) [reply src/common/Collision/Management/VMapManager2.cpp:169]: Look all changes

**Comment by FrancescoBorzi** [comment]
@azerothcore/testers @sedatyf can anyone check this please?

**Comment by Sedatyf** [comment]
> @azerothcore/testers @Sedatyf can anyone check this please?

Sorry, I forgot to mention it but I'll test as soon as I get time as it seems it will be a bit long to test

**Comment by Sedatyf** [comment]
I'll test this PR today or at least this weekend. Do I have to generate again all maps in order to test it? (mmaps, vmaps, etc.) Or do I just need to pull this PR and do usual steps?

**Comment by Winfidonarleyan** [comment]
> I'll test this PR today or at least this weekend. Do I have to generate again all maps in order to test it? (mmaps, vmaps, etc.) Or do I just need to pull this PR and do usual steps?

No need re extract maps

**Comment by Sedatyf** [comment]
Hey @Winfidonarleyan, so we tried to clip through wall, we casts fear on each other, and nothing really interesting came up. We tested what you linked on your PR and everything works like a charm.
We played like nearly an hour without an issue.
Please tell me if you see some test I could do, otherwise, it seems fine to me

**Comment by Winfidonarleyan** [comment]
Very very good @Sedatyf, thanks for test, i think that's enough.

**Review by Yehonal — APPROVED** [review]

**Comment by Helias** [comment]
merge conflicts

**Comment by mpfans** [comment]
I have a question. Why delete Blizzard addons?
